src/components/DateRangePicker.js

```diff
@@ -83,76 +83,31 @@ const DateRangePicker = React.createClass({
     }
   },
 
-  _handleDateSelect (date) {
-    const selectedStartDate = this.props.selectedStartDate;
-    const selectedEndDate = this.props.selectedEndDate;
-    let updatedDates = [null, null];
-
-    updatedDates = this._deselectDate(date, selectedStartDate, selectedEndDate);
-    updatedDates = updatedDates || this._startAndEndDate(date, selectedStartDate, selectedEndDate);
-    updatedDates = updatedDates || this._noEndDate(date, selectedStartDate, selectedEndDate);
-    updatedDates = updatedDates || this._noStartDate(date, selectedStartDate, selectedEndDate);
-    updatedDates = updatedDates || this._noStartOrEndDate(date, selectedStartDate, selectedEndDate);
-
-    this.props.onDateSelect(updatedDates[0], updatedDates[1]);
+  _endDateIsBeforeStartDate (startDate, endDate) {
+    return moment.unix(endDate).isBefore(moment.unix(startDate));
   },
 
-  _startAndEndDate (selected, start, end) {
-    let updatedDates;
-
-    if (start && end) {
-      if (selected < start && selected < end) {
-        updatedDates = [selected, end];
-      } else if ((selected > start && selected < end) || selected > end) {
-        updatedDates = [start, selected];
-      }
-    }
-
-    return updatedDates;
-  },
-
-  _noEndDate (selected, start, end) {
-    let updatedDates;
-
-    if (start && !end) {
-      if (selected > start) {
-        updatedDates = [start, selected];
-      } else {
-        updatedDates = [selected, start];
-      }
-    }
-
-    return updatedDates;
-  },
-
-  _noStartDate (selected, start, end) {
-    let updatedDates;
-
-    if (!start && end) {
-      if (selected < end) {
-        updatedDates = [selected, end];
-      } else {
-        updatedDates = [end, selected];
-      }
+  _handleDateSelect (date) {
+    let endDate;
+    let startDate;
+    const existingRangeComplete = this.props.selectedStartDate && this.props.selectedEndDate;
+    const existingRangeEmpty = !this.props.selectedStartDate && !this.props.selectedEndDate;
+
+    if (existingRangeComplete || existingRangeEmpty) {
+      startDate = date;
+      endDate = null;
+    } else {
+      startDate = this.props.selectedStartDate;
+      endDate = date;
     }
```

Comment:
This doesn't allow a user to expand the date range either. Are we cutting that out too?